actions/docs-verifier/Directory.Build.props

```diff
@@ -3,15 +3,15 @@
 
   <PropertyGroup>
     <Nullable>enable</Nullable>
-    <LangVersion>9</LangVersion>
+    <LangVersion>10</LangVersion>
     <EnableNETAnalyzers>true</EnableNETAnalyzers>
-    <AnalysisMode>AllEnabledByDefault</AnalysisMode>
-    <AnalysisLevel>latest</AnalysisLevel>
+    <AnalysisLevel>latest-All</AnalysisLevel>
     <EnforceCodeStyleInBuild>true</EnforceCodeStyleInBuild>
+    <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.CodeAnalysis.NetAnalyzers" Version="5.0.3" />
+    <PackageReference Include="Microsoft.CodeAnalysis.NetAnalyzers" Version="7.0.0-preview1.22068.2" />
```

Comment:
I think we should probably stay on .NET 6 (LTS), as it is stable.

Reply:
@IEvangelist It has a false positive for top-level statements. I could suppress the warning, but I preferred a preview version as long as it doesn't have any problems.

Reply:
I'm fine with it, but let's add a comment inline stating that so it doesn't come off as a mistake.

```xml
<!-- This preview version fixes a false positive bug with top-level statements. -->
```

Reply:
```suggestion
    <!-- This preview version fixes a CA1812 (Avoid uninstantiated internal classes) false positive bug with top-level statements. -->
    <PackageReference Include="Microsoft.CodeAnalysis.NetAnalyzers" Version="7.0.0-preview1.22068.2" />
```